server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -3401,4 +3402,19 @@ public void testSupplyTombstoneDoc() throws Exception {
 
         closeShards(shard);
     }
+
+    public void testResetEngine() throws Exception {
```

Comment:
can we also have a test that checks we don't reset if we don't need to?

Reply:
I added this check to IndexShardTests#testRollbackReplicaEngineOnPromotion.